vllm/entrypoints/chat_utils.py

```diff
@@ -1374,5 +1374,22 @@ def apply_mistral_chat_template(
             "template")
         raise ValueError(str(e)) from e
 
-def random_tool_call_id() -> str:
-    return f"chatcmpl-tool-{random_uuid()}"
+
+def get_history_tool_calls_cnt(conversation: list[ConversationMessage]):
+    idx = 0
+    for msg in conversation:
+        if msg['role'] == 'assistant':
+            tool_calls = msg.get('tool_calls')
+            if tool_calls is not None:
+                for _ in tool_calls:
```

Comment:
Can we count the length of `tool_calls` here?

Reply:
tool call here would always be a list, so I don't think the annotations is correct. Let's use `len` and avoid using double for loop